content/outcomes/cloud-native-development-in-practice/configure-app.md

```diff
@@ -0,0 +1,454 @@
+    There is more [detailed documentation](https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#expressions-beandef)
+    on using Spring expressions in bean definitions and annotations,
+    if you are interested.
+
+1.  If you get stuck,
+    review the solution:
+
+    ```bash
+    git show configuration-solution:src/main/java/io/pivotal/pal/tracker/WelcomeController.java
+    ```
+
+## Verify it works
+
+Run your app with the `bootRun` Gradle task.
+It will fail to start because Spring Boot is unable to find a value for
+the requested variable.
+
+From now on the `WELCOME_MESSAGE` environment variable must be set to
+run your app.
+One way to do this is to add the environment variable assignment before
+the Gradle command.
+
+```bash
+WELCOME_MESSAGE=howdy ./gradlew bootRun
+```
+
+Notice you are using a different welcome message than the previous
+statically configured message.
+
+### Why is the `@Value` name `welcome.message`?
+
+You may be wondering why the name given in the `@Value` annotation
+is `welcome.message` and not the name of the environment variable,
+`WELCOME_MESSAGE`.
+
+Spring can take configuration information from
+[many different sources](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config).
+In order to support this, and to have a consistent way of naming
+configuration items from different sources, Spring uses what is known
+as [relaxed binding](https://github.com/spring-projects/spring-boot/wiki/Relaxed-Binding-2.0).
+Among other things, this defines the rules that convert environment
+variable names, such as `WELCOME_MESSAGE`, to the common format represented
+by `welcome.message`.
+This means that you could later change the source of the welcome message
+text to, for example, a Java property without having to modify the code.
+
+## Manage local configuration
+
+Running your application like this with environment variables in the
+command line every time is tedious.
+
+You can [configure your Gradle build](https://cloudnative.tips/configuring-a-java-application-for-local-development-60e2c9794ca7)
+to make this easier.
+
+1.  Extend the `bootRun` and `test` tasks to set the required
+    environment variable by adding the following to the end of your
+    `build.gradle` file:
+
+    ```groovy
+    bootRun.environment([
+        "WELCOME_MESSAGE": "howdy",
+    ])
+
+    test.environment([
+        "WELCOME_MESSAGE": "Hello from test",
+    ])
+    ```
+
+1.  You can review the solution here:
+
+    ```bash
+    git show configuration-solution:build.gradle
+    ```
+
+1.  This will instruct Gradle to set that environment variable for you
+    when you run the `bootRun` task, so you can go back to just using:
+
+    ```bash
+    ./gradlew bootRun
+    ```
+
+This has the added benefit of documenting required environment variables
+and supporting multiple operating systems.
+
+## TAS Environment Variables
```

Comment:
```suggestion
## Tanzu Application Service Environment Variables
```
I know we do it all the time, but for public documentation/blogs/guides in general, VMware prefers not to "market" products by their TLA's. This is not consistent across VMware, and you can find counter examples (including likely in TDC), but we want to limit it as much as is reasonably possible. Scan through these doc and replace as you find them. I'll try to highlight too.